monitor/inclusion_checker.go

```diff
@@ -0,0 +1,156 @@
+package monitor
+
+import (
+	"context"
+	"crypto/sha256"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/google/certificate-transparency-go"
+	"github.com/google/certificate-transparency-go/tls"
+	"github.com/jmhodges/clock"
+	"github.com/letsencrypt/ct-woodpecker/storage"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+var oldestUnseen = promauto.NewGauge(prometheus.GaugeOpts{
+	Name: "oldest_unincorporated_cert",
+	Help: "Number of seconds since the oldest unincorporated certificate was submitted",
+})
+
+type InclusionOptions struct {
+	Interval       time.Duration
+	FetchBatchSize int64
+}
+
+type inclusionChecker struct {
+	logger           *log.Logger
+	client           monitorCTClient
+	logURI           string
+	db               storage.Storage
+	signatureChecker *ct.SignatureVerifier
+	clk              clock.Clock
+	logID            int64
+
+	interval  time.Duration
+	batchSize int64
+}
+
+func (ic *inclusionChecker) run() {
```

Comment:
Can you add a graceful `stop()` to the `inclusionChecker` and model the `run` sleep after the interruptable method the `certSubmitter` and `sthFetcher` are using now? Thanks!